server/core/src/https/v1.rs

```diff
@@ -2423,9 +2411,24 @@ pub async fn domain_attr_put(
     Extension(kopid): Extension<KOpId>,
     VerifiedClientInformation(client_auth_info): VerifiedClientInformation,
     Path(attr): Path<String>,
-    Json(values): Json<Vec<String>>,
+    Json(values): Json<Vec<serde_json::Value>>,
 ) -> Result<Json<()>, WebError> {
     let filter = filter_all!(f_eq(Attribute::Class, EntryClass::DomainInfo.into()));
+
+    let values: Vec<String> = values
+        .into_iter()
+        .map(|v| {
+            #[allow(clippy::expect_used)]
+            if v.is_boolean() {
+                v.as_bool()
+                    .map(|b| b.to_string())
+                    .expect("Failed to un-wrap thing we already identified as a boolean!")
+            } else {
+                v.as_str().unwrap().to_string()
```

Comment:
I'm not sure I like this, what's the intent here? It should work with strings because we convert in a lower stage from str to bool.

Reply:
Yeah I'm not a huge fan either. At some point we have to turn it into a string, but if we don't do it here, then it throws a deser issue on the receiever because it's expecting `Vec<String>` not `Vec<string or bool or other rando things>`

None of this was being caught because all the current tests only send string-type-values for attributes.

Reply:
JSON is "unambiguous" so this means if we have a bool value, that the client didn't str encode it. https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html#method.is_boolean

So that means the issue is actually:

https://github.com/kanidm/kanidm/blob/master/libs/client/src/lib.rs#L1944

Which should have enable.to_string() instead of vec![enable].

Reply:
yeah, I was trying to take _**actual JSON"**_ 'cause that's nicer from the client side. oh wells.